MobileManager/Controllers/AppPackageController.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MobileManager.Controllers.Interfaces;
+using MobileManager.Http.Clients.Interfaces;
+using MobileManager.Logging.Logger;
+using MobileManager.Models.App;
+using Newtonsoft.Json;
+
+#pragma warning disable 1998
+
+namespace MobileManager.Controllers
```

Comment:
Controller implementation should not contain direct work with I/O. Create a separate service for working with AppPackages. 
Also a repository for md5 hash would be much better then lookup for files. I would go for creating a repository class and DB table for storing AppPackage object and work with that rather then working directly with files.

Reply:
I agree this Controllers do too many logic in one place and it should be separated to some logic units. Good point.
But adding DB table seems to me like unnecessary over-complication. Files on disk are special kind of database and you don't need to solve any inconsistency between your DB and files. Just work with files seems ok to me. At least for now.

Reply:
the database is only for info about the files stored on disk - it allows fast search for stored files and can help with future problems regarding performance - I/O operations are really slow, its much better to work with some DB info (like looking up all stored files, controlling hash values and etc.).

Reply:
the database is only for info about the files stored on disk - it allows fast search for stored files and can help with future problems regarding performance - I/O operations are really slow, its much better to work with some DB info (like looking up all stored files, controlling hash values and etc.).

Reply:
nevertheless, it's up to you - the database is already there, sounds to me like an easy win in this case.

Reply:
All discussed changes done, please check new version.